### PR TITLE
refactor: Management and creation of favorited nodes

### PR DESCRIPTION
**Pull request by traeok**
## Proposed changes

Refactors the favorited node initialization, management, and importing procedure from local storage.

## Release Notes

Milestone: 3.0

Changelog: 

- Refactored the initialization, management, and importing procedure for favorited nodes in Zowe Explorer.

## Types of changes

What types of changes does your code introduce to Zowe Explorer?
_Put an `x` in the boxes that apply_

- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Updates to Documentation or Tests (if none of the other choices apply)

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This checklist will be used as reference for both the contributor and the reviewer_

- [x] I have read the [CONTRIBUTOR GUIDANCE](https://github.com/zowe/zowe-explorer-vscode/wiki/Best-Practices:-Contributor-Guidance) wiki
- [x] PR title follows [Conventional Commits Guidelines](https://www.conventionalcommits.org/en/v1.0.0-beta.2/)
- [x] PR Description is included
- [ ] gif or screenshot is included if visual changes are made
- [x] `yarn workspace vscode-extension-for-zowe vscode:prepublish` has been executed
- [ ] All checks have passed (DCO, Jenkins and Code Coverage)
- [ ] I have added unit test and it is passing
- [ ] I have added integration test and it is passing
- [ ] There is coverage for the code that I have added
- [x] I have tested it manually and there are no regressions found

### CODE REVIEW

**Comment by traeok** [comment]
Feel free to review this when convenient, but no need to merge once it has the required approvals - I plan on updating the base branch to `next` once the FileSystemProvider is merged.

**Review by t1m0thyj — COMMENTED** [review]
Looks pretty good, thanks @traeok! Noticed weird behavior in a few cases when the same item is open both in favorites and outside of favorites:
* For a USS directory, when creating a new file inside, the list of children only updates in one place and isn't synchronized
* For a PDS, when renaming one of its members, the list of children only updates in one place and the other place still shows the ~~old name~~ (with strikethrough)
* For a USS directory, when renaming one of its children, the rename operation happens twice and fails the second time

File: packages/zowe-explorer/CHANGELOG.md
Comment (traeok) [reply packages/zowe-explorer/CHANGELOG.md:15]: In hindsight I was definitely being verbose here - sorry, bad habit 😋 

I rephrased it to "Refactored behavior and management of Favorites in Zowe Explorer" since nodes are a bit too developer-specific for the primary changelog. 

In this branch I've also adjusted the UX of session favorites to be consistent for each tree, so considering this and the state synchronization, I added "behavior" so users aren't surprised by some changes that weren't listed.

**Comment by traeok** [comment]
Thanks @t1m0thyj, I've updated the code to handle the mentioned scenarios. 

- **USS directory 'Create File' issue**: Was missing a refresh for the favorited parent; resolved in c7a305d
- **PDS rename issue**: This was also missing a refresh (for sorting); resolved in f84d7bd
- **USS rename issue**: `findFavoritedNode` had an issue where it was considering sessions before favorited directories/files; fixed logic for that function in 02f23ea and updated some of the rename logic

**Review by t1m0thyj — APPROVED** [review]
LGTM, thanks @traeok!

When testing the latest round of changes, I opened a favorited USS directory both as a favorite and non-favorite node, then renamed a file inside the non-favorited node and ZE updated the name only for the favorited instance and not the non-favorited one. This happened only once and I haven't been able to reproduce it since, so I'm good with approving 🙂

**Review by JillieBeanSim — CHANGES_REQUESTED** [review]
Thanks @traeok for working on this refactor, I do see an error when trying to access the favorited job on this branch but not the next branch
`next`:
Screenshot 2024-04-25 at 8 58 42 AM
(looks like the fix that went in for the `undefined` needs porting)
`refactor/favorites`:
Screenshot 2024-04-25 at 8 50 34 AM

**Comment by traeok** [comment]
> Thanks @traeok for working on this refactor, I do see an error when trying to access the favorited job on this branch but not the next branch `next`

Thanks for spotting this Billie! I added a check to the jobs provider and removed an extra context value assignment in b7c76e6 that should resolve the issue 🙂

**Review by JillieBeanSim — DISMISSED** [review]
thanks for pushing the fix, LGTM!

**Review by t1m0thyj — APPROVED** [review]
LGTM, thanks @traeok for fixing the favorite profiles icons!

File: packages/zowe-explorer-api/src/tree/IZoweTreeNode.ts
Comment (zFernand0) [inline packages/zowe-explorer-api/src/tree/IZoweTreeNode.ts:135]: general question:
Do we plan to remove all these deprecated methods before V3, or as part of a future V4?

**Review by zFernand0 — APPROVED** [review]
LGTM! 😋 
Functionality works great! 🥳

File: packages/zowe-explorer-api/src/tree/IZoweTreeNode.ts
Comment (traeok) [reply packages/zowe-explorer-api/src/tree/IZoweTreeNode.ts:135]: This will be for v4 to give extenders enough time to transition 😋

**Pull request merged**
